Homework_6/commentify/frontend/src/components/Main/Main.js

```diff
@@ -0,0 +1,82 @@
+import React, { Component } from "react";
+import { Header, Posts, Panel } from "../../components";
+import { fetchPosts } from "../../Api/api";
+import calculateAverages from "../../helpers/average";
+
+class Main extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: true,
+      pool: [],
+      allPosts: [],
+      searchedNotFound: false,
+    };
+  }
+
+  componentDidMount() {
+    fetchPosts()
+      .then((response) =>
+        this.setState({
+          loading: false,
+          allPosts: calculateAverages(response.data),
+          pool: calculateAverages(response.data),
+        })
+      )
+      .catch((error) => console.error("Error fetching posts:", error));
+  }
+
+  disableEnablePost = (id, status) => {
+    this.setState((prevState) => {
+      const index = this.state.allPosts.findIndex((post) => id === post.id);
+      const newPosts = [...prevState.allPosts];
+      newPosts[index] = {
+        ...newPosts[index],
+        selected: status,
+      };
+      return { ...prevState, allPosts: newPosts };
+    });
+  };
+
+  filterBySearch = (value) => {
+    const comIncludes = (coms, value) => {
+      for (let i = 0; i < coms.length; i++) {
+        if (coms[i].comment.toLowerCase().includes(value.toLowerCase())) {
```

Comment:
what if you just return coms[i].comment.toLowerCase().includes(value.toLowerCase()) ?